crates/storage/provider/src/writer/database.rs

```diff
@@ -17,12 +17,10 @@ where
         &mut self,
         first_tx_index: TxNumber,
         _: BlockNumber,
-        receipts: Vec<Option<Receipt>>,
```

Comment:
so you mean that for this specific function we should keep `Vec<Option<Receipt>>` or does it affect also other methods? In summary you want to keep this behavior right? (Was thinking that this was equivalent to simply have no receipt)

> If `receipt` is `None`, it has been pruned.

Reply:
@joshieDo This means also that in `ExecutionOutcome`, we cannot use `Receipts` directly here no?

https://github.com/paradigmxyz/reth/blob/af5ae5a792762d2b02abc31a99807ed185c3b617/crates/evm/execution-types/src/execution_outcome.rs#L44

Because with `Receipts`, we have a `vec[vec[Receipt]]` but here you mention the need for `vec[vec[Option<Receipt>]]` due to staged sync pruning flags no?

Reply:
Yes the initial goal of this PR was indeed to replace the `Receipts` type with its alloy equivalent but as discussed here https://github.com/alloy-rs/alloy/pull/1247#pullrequestreview-2287495139, in alloy we no longer have an `Option` around the `Receipt` natively inside `Receipts`.

To try to do things little by little I just created another PR https://github.com/paradigmxyz/reth/pull/12162 to minimize the changes with:

```rust
pub type Receipts = alloy_consensus::Receipts<Option<Receipt>>;
```

What I propose:
- We merge the first small PR that eliminates the implementation of `Receipts` in reth https://github.com/paradigmxyz/reth/pull/12162.
- We close this PR (where we are discussing right now) which will then become obsolete.
- We open other small issues to try to better prune and add unit tests according to https://github.com/alloy-rs/alloy/pull/1247#pullrequestreview-2287495139
- Gradually we try to remove the `Option` from `Receipts`

cc @joshieDo @mattsse @shekhirin

Reply:
@joshieDo As you prefer, I think both ways are fine:
- We could maybe first merge https://github.com/paradigmxyz/reth/pull/12162 first so that we can play easier with the `Option` or not because the alloy `Receipts<T>` has the generic type T so that maybe we can replace step by step.
- Also it would maybe allow to directly focus on alloy and not cause additional regression on reth.
- But if you don't agree no worries, we can first fix the pruning and then merge https://github.com/paradigmxyz/reth/pull/12059